### PR TITLE
Added a guide for Self-Hosting Hoppscotch

### PR DESCRIPTION
**Pull request by SanskritiHarmukh**
This PR contains the following:

 - [x] A guide for self-hosting Hoppscotch based on the Community Edition.
 - [x] Updated the Workspaces section in the [Improve Your API Workflow guide](https://docs.hoppscotch.io/guides/articles/improving-your-api-workflow#workspaces).

### CODE REVIEW

**Review by thetronjohnson — APPROVED** [review]

File: guides/articles/self-host-hoppscotch-on-your-own-servers.mdx [hunk 1]
```diff
@@ -0,0 +1,283 @@
+---
+sidebarTitle: Self-Host Hoppscotch on your own servers
+title: Self-Host Hoppscotch on your own servers
+description: Set up Hoppscotch on your servers for complete control and customization.
+---
+
+Self-hosting Hoppscotch gives you complete control over your API development workflow and allows you to deploy Hoppscotch in your own data center or cloud, giving you greater control over data and security.
+
+Self-hosted Hoppscotch comes in two variants - [Community](https://docs.hoppscotch.io/documentation/self-host/community-edition/getting-started) and [Enterprise Edition](https://docs.hoppscotch.io/documentation/self-host/enterprise-edition/getting-started), both of which can be deployed on systems that support Docker. You can host Hoppscotch on your servers, providing a workspace that is private to the individuals or teams using Hoppscotch.
```
Comment (balub) [inline guides/articles/self-host-hoppscotch-on-your-own-servers.mdx:9]:
```suggestion
Self-hosted Hoppscotch comes in two variants - [Community](https://docs.hoppscotch.io/documentation/self-host/community-edition/getting-started) and [Enterprise Edition](https://docs.hoppscotch.io/documentation/self-host/enterprise-edition/getting-started), both of which can be deployed on systems that support Docker. You can host Hoppscotch on your servers, providing a private workspace for the individuals or teams using it.
```

File: guides/articles/self-host-hoppscotch-on-your-own-servers.mdx [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+
+This guide covers the basics of self-hosting Hoppscotch, including the necessary configurations and settings you need to get started.
```
Comment (balub) [inline guides/articles/self-host-hoppscotch-on-your-own-servers.mdx:11]:
```suggestion
This guide covers the basics of self-hosting Hoppscotch, including the configurations and settings needed to get started
```

File: guides/articles/self-host-hoppscotch-on-your-own-servers.mdx [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+
+## Pre-requisites
+
+Before you start self-hosting Hoppscotch, ensure that your system or environment meets the following requirements:
```
Comment (balub) [inline guides/articles/self-host-hoppscotch-on-your-own-servers.mdx:15]:
```suggestion
Before you start ensure that your system or environment meets the following requirements:
```

File: guides/articles/self-host-hoppscotch-on-your-own-servers.mdx [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+VITE_APP_PRIVACY_POLICY_LINK=https://docs.hoppscotch.io/support/privacy
+
+# Set to `true` for subpath based access
+ENABLE_SUBPATH_BASED_ACCESS=false
+
+```
+
+### 1. Database Configuration
+
+By default, Hoppscotch ships with a Docker container that has a pre-configured Postgres database. If you prefer to use your own Postgres database, either on the cloud or locally, you’ll need to set up a valid database URL with the correct credentials.
```
Comment (balub) [inline guides/articles/self-host-hoppscotch-on-your-own-servers.mdx:111]: We dont ship with a pre-configured postgres database, why are we adding this here ?

File: guides/articles/self-host-hoppscotch-on-your-own-servers.mdx [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+Similarly, you can follow the specific setup instructions for other OAuth providers to complete your configuration.
+
+### 4. Subpath Access
+
+Subpath access allows you to host multiple services under a single domain by assigning each service a specific subpath.
+
+To enable subpath-based access, set `ENABLE_SUBPATH_BASED_ACCESS=true` (the **default** setting is **false**).
+
+When enabled, here's how it affects your setup:
+
```
Comment (balub) [inline guides/articles/self-host-hoppscotch-on-your-own-servers.mdx:207]: Lets just replace this whole section and just say if subpath is enabled you can access the 3 services on the same domain but different routes and if its not enabled you access them on different ports.

**Review by balub — CHANGES_REQUESTED** [review]

File: guides/articles/self-host-hoppscotch-on-your-own-servers.mdx (hunk 4 above)
Comment (SanskritiHarmukh) [reply guides/articles/self-host-hoppscotch-on-your-own-servers.mdx]: My bad

File: guides/articles/self-host-hoppscotch-on-your-own-servers.mdx (hunk 5 above)
Comment (SanskritiHarmukh) [reply guides/articles/self-host-hoppscotch-on-your-own-servers.mdx]: Done

**Review by balub — APPROVED** [review]
LGTM 🚀

**Pull request merged**